packages/core/src/storages/request_queue_v2.ts

```diff
@@ -53,7 +53,8 @@ const RECENTLY_HANDLED_CACHE_SIZE = 1000;
  * @category Sources
  */
 export class RequestQueue extends RequestProvider {
-    private _listHeadAndLockPromise: Promise<void> | null = null;
+    private listHeadAndLockPromise: Promise<void> | null = null;
+    private queueHasLockedRequests: boolean | undefined = undefined;
 
     /**
      * Returns `true` if there are any requests in the queue that were enqueued to the forefront.
```

Comment:
Can you please update this function regarding notes in 
https://github.com/apify/apify-core/issues/19218#issuecomment-2621466723 ?

Reply:
I think I managed to make it work regardless of whether or not there are multiple clients.